shell/platform/android/io/flutter/embedding/engine/systemchannels/AccessibilityChannel.java

```diff
@@ -0,0 +1,101 @@
+package io.flutter.embedding.engine.systemchannels;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.HashMap;
+
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.BasicMessageChannel;
+import io.flutter.plugin.common.StandardMessageCodec;
+
+public class AccessibilityChannel {
+  public BasicMessageChannel<Object> channel;
+  private AccessibilityMessageHandler handler;
+
+  private final BasicMessageChannel.MessageHandler<Object> parsingMessageHandler = new BasicMessageChannel.MessageHandler<Object>() {
+    @Override
+    public void onMessage(Object message, BasicMessageChannel.Reply<Object> reply) {
+      // If there is no handler to respond to this message then we don't need to
+      // parse it. Return.
+      if (handler == null) {
+        return;
+      }
+
+      @SuppressWarnings("unchecked")
+      final HashMap<String, Object> annotatedEvent = (HashMap<String, Object>) message;
+      final String type = (String) annotatedEvent.get("type");
+      @SuppressWarnings("unchecked")
+      final HashMap<String, Object> data = (HashMap<String, Object>) annotatedEvent.get("data");
+
+      switch (type) {
+        case "announce":
+          handler.announce((String) data.get("message"));
+          break;
+        case "tap": {
+          Integer nodeId = (Integer) annotatedEvent.get("nodeId");
+          if (nodeId != null) {
+            handler.onTap(nodeId);
+          }
+          break;
+        }
+        case "longPress": {
+          Integer nodeId = (Integer) annotatedEvent.get("nodeId");
+          if (nodeId != null) {
+            handler.onLongPress(nodeId);
+          }
+          break;
+        }
+        case "tooltip": {
+          handler.tooltip((String) data.get("message"));
+          break;
+        }
+        // Requires that the node id provided corresponds to a live region, or TalkBack will
+        // ignore the event. The event will cause talkback to read out the new label even
+        // if node is not focused.
+        case "updateLiveRegion": {
```

Comment:
updateLiveRegion is removed in master and otherwise soft-deprecated. We shouldn't add it here

Reply:
I'm curious, how do we handle deprecations of things like this?  Before now I don't think there was an explicit API representing the message, so there's no place to even add an @Deprecated, right?

Reply:
Since different platforms have different accessibility features, the API guidelines are "if you do X then the platform _may_ do Y".  It's not really possible to depend on these APIs because there is no observable effect (in the code) and requires users to have enabled a separate setting.